tests/antool/test_syscalls.c

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright 2017, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * test_syscalls.c -- functional tests for vltrace
+ */
+
+#define _GNU_SOURCE
+#define N_WORKERS 10
+
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <poll.h>
+#include <signal.h>
+#include <assert.h>
+#include <pthread.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/syscall.h>
+#include <sys/utsname.h>
+#include <sys/time.h>
+#include <sys/ioctl.h>
+#include <sys/epoll.h>
+#include <sys/xattr.h>
+#include <sys/file.h>
+#include <sys/socket.h>
+#include <sys/inotify.h>
+#include <sys/mman.h>
+#include <sys/mount.h>
+#include <sys/select.h>
+#include <sys/swap.h>
+#include <sys/sendfile.h>
+
+#include <linux/futex.h>
+#include <linux/fs.h>
+#include <linux/falloc.h>
+
+#define F_ADD_SEALS		1033
+#define F_GET_SEALS		1034
+#define ANY_STR			"any-string"
+
+#define PATTERN_START		((int)0x12345678)
+#define PATTERN_END		((int)0x87654321)
+#define BUF_SIZE		0x100
+
+#define MARK_START()		close(PATTERN_START)
+#define MARK_END()		close(PATTERN_END)
+
+static int counter;
+
+/*
+ * s -- sleep a while
+ */
+static void
+s()
+{
+#define NSLEEP 1000000
```

Comment:
Please remove this macro, it is clearly used in a single place.
BTW, this routine does not sleep, it is rather busy.